vmware-base/src/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -1184,4 +1206,261 @@ public ManagedObjectReference waitForPortGroup(String networkName, long timeOutM
+     * @param gridKGPUMemory (Video RAM of each GPU in the card)
+     * @throws Exception
+     */
+    private void updateGpuCapacities(final HashMap<String, VgpuTypesInfo> gpuCapacity, final String groupName, final long countGridKSharedGPUs, final List<HostGraphicsInfo> graphicsInfo, final List<String> sharedPassthruGpuTypes, final long gridKGPUMemory) throws Exception {
+        /*
+         * 0 - grid_k100 or grid_k200
+         * 1 - grid_k120q or grid_k220q
+         * 2 - grid_k140q or grid_k240q
+         * 3 - grid_k160q or grid_k260q
+         * 4 - grid_k180q or grid_k280q
+         */
+        final long remainingCapacities[] = new long[5];
+
+        remainingCapacities[0] = 8l * countGridKSharedGPUs;
+        remainingCapacities[1] = 8l * countGridKSharedGPUs;
+        remainingCapacities[2] = 4l * countGridKSharedGPUs;
+        remainingCapacities[3] = 2l * countGridKSharedGPUs;
+        remainingCapacities[4] = countGridKSharedGPUs;
+
+        for (final HostGraphicsInfo graphicInfo : graphicsInfo) {
+            if (graphicInfo.getDeviceName().equals(groupName) && graphicInfo.getGraphicsType().equals("shared")) {
+                if (CollectionUtils.isNotEmpty(graphicInfo.getVm())) {
+                    String vgpuType = "None";
+
+                    for (ManagedObjectReference mor : graphicInfo.getVm()) {
+                        final VirtualMachineMO vmMo = new VirtualMachineMO(_context, mor);
+
+                        if (vgpuType.equals("None") && vmMo != null && vmMo.getConfigInfo() != null && vmMo.getConfigInfo().getHardware() != null) {
+                            final List<VirtualDevice> devices = vmMo.getConfigInfo().getHardware().getDevice();
+
+                            for (VirtualDevice device : devices) {
+                                if (device instanceof VirtualPCIPassthrough) {
+                                    if (device.getBacking() != null && (device.getBacking() instanceof VirtualPCIPassthroughVmiopBackingInfo)) {
+                                        final VirtualPCIPassthroughVmiopBackingInfo backingInfo = (VirtualPCIPassthroughVmiopBackingInfo) device.getBacking();
+
+                                        if (backingInfo.getVgpu() != null) {
+                                            vgpuType = backingInfo.getVgpu();
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    // If GRID K1, then search for only K1 vGPU types. Same for GRID K2.
+                    // The remaining capacity of one type affects other vGPU type capacity.
+                    // Each GPU should always contain one type of vGPU VMs.
+                    if ((groupName.equals("NVIDIAGRID K1") && vgpuType.equals("grid_k100")) || (groupName.equals("NVIDIAGRID K2") && vgpuType.equals("grid_k200"))) {
```

Comment:
@nitin-maharana can you refactor this code by assigning the groupName against an index where `graphicInfo.getVm().size()` should be used instead of static/hard-coded `remainingCapacities` values.

Reply:
@rhtyd, I didn't get you properly. I guess you are saying to replace the multiple remainingCapacities assignments present in the if..else statement with a single assignment using index variable by matching the grid type. Am I correct?

Note:
Actually, the reason why I hardcoded those assignments because unlike XenServer, VMware doesn't provide libraries to get the remaining capacities of each vGPU type. To obtain that I used the VM size, which says how many VMs of the particular vGPU type present in the card. Accordingly, it reduces the capacity of each vGPU type(One vGPU type affects the capacity of other vGPU types).

Reply:
@nitin-maharana my suggestion is around refactoring, all the ifs specific to a graphics card type/name could be wrapped to a class where the behavior settings/values could be moved. It will make it easier in future to add more graphics cards, maintain code etc.